src/plonk/circuit.rs

```diff
@@ -1,10 +1,13 @@
 use core::cmp::max;
 use core::ops::{Add, Mul};
 use ff::Field;
-use std::convert::TryFrom;
+use std::{
+    convert::TryFrom,
+    ops::{Neg, Sub},
+};
 
 use super::{lookup, permutation, Error};
-use crate::poly::Rotation;
+use crate::{arithmetic::FieldExt, poly::Rotation};
```

Comment:
Where is `FieldExt` needed?

Reply:
It's used in the type bounds of the `Neg` and `Sub` impls for `Expression<F>`, since we need the ability to define `-F::one()`.

Reply:
@ebfull clarified that he meant this could just be `F: Field`.